### PR TITLE
Fix configargparse 0.12.0

### PR DESCRIPTION
**Pull request by bmw**
Built on #4650.

Because we define the `ShowWarning` class inside the function, we actually are adding multiple different `ShowWarning` classes to the object. This doesn't hurt anything besides performance and I started to fix it but we should only apply the minimal patch for now. This is not a behavior change.

### CODE REVIEW

**Comment by bmw** [comment]
I modified this PR so tests will pass on the 0.14.x branch. I also added an additional commit to clean up the code here a bit, however, I don't plan on including it in 0.14.1.

**Comment by bmw** [comment]
Can I get another review on this? I changed some things later on Friday and I've set up GitHub to dismiss stale reviews automatically.

File: certbot/tests/util_test.py [hunk 1]
```diff
@@ -368,6 +368,29 @@ def test_help(self):
...
 
+    def test_set_constant(self):
+        """Test when ACTION_TYPES_THAT_DONT_NEED_A_VALUE is a set.
+
+        This variable is a set in configargparse versions < 0.12.0.
+
+        """
+        self._test_constant_common(set)
+
+    def test_tuple_constant(self):
```
Comment (ohemorange) [inline certbot/tests/util_test.py]: the comment needs switching from `set` to `tuple`

Comment (bmw) [inline certbot/tests/util_test.py]: fixed

Comment (ohemorange) [inline certbot/tests/util_test.py]: one more

Comment (bmw) [inline certbot/tests/util_test.py]: no more

**Pull request merged**